code/game/objects/structures/mirror.dm

```diff
@@ -27,8 +28,22 @@
 
 /obj/structure/mirror/Initialize(mapload)
 	. = ..()
-	if(icon_state == "mirror_broke" && !broken)
-		atom_break(null, mapload)
+	if(icon_state == "mirror_broke")
+		if(!broken)
+			atom_break(null, mapload)
+	else
+		add_reflection_comp()
+
+/obj/structure/mirror/Destroy()
+	QDEL_NULL(reflection_comp)
+	return ..()
+
+/obj/structure/mirror/proc/add_reflection_comp()
+	if(reflection_comp)
+		return
+	var/list/mask_filter = alpha_mask_filter(icon = icon('icons/obj/watercloset.dmi', "mirror_mask"))
+	var/matrix/matrix = matrix(-0.75, 0, 0, 0, 0.75, 0)
+	reflection_comp = AddComponent(/datum/component/reflection, use_parent_dir = TRUE, reflection_filter = mask_filter, reflection_matrix = matrix)
```

Comment:
why are you storing a var to the component? you dont appear to be doing anything to it other than deleting it on Destroy() and that behavior should be done by the component

Reply:
The component is also removed when the mirror is broken and readded once it's been fixed. See lines 110 and 139.

Reply:
You should use LoadComponent for the fixing and use signals to react to the breaking.
You can handle the signal stuff in the mirror and just have a delete proc to register onto, but you shouldn't be storing the component like a var.